cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -121,11 +121,32 @@ func (m *McmManager) GetMachineDeploymentForMachine(machine *Ref) (*MachineDeplo
 		//Considering the possibility when Machine has been deleted but due to cached Node object it appears here.
 		return nil, fmt.Errorf("Node does not Exists")
 	}
-	machStr := strings.Split(machine.Name, "-")
-	machName := strings.Join(machStr[:len(machStr)-2], "-")
+	machineObject, err := m.machineclient.Machines(m.namespace).Get(machine.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to fetch Machine object %s %+v", machine.Name, err)
+	}
+
+	var machineSetName, machineDeploymentName string
+	if len(machineObject.OwnerReferences) > 0 {
+		machineSetName = machineObject.OwnerReferences[0].Name
```

Comment:
I hope there won't be a situation with multiple owner references. Otherwise looks good.

Reply:
It could actually be, as it is also defined as an array.